src/System.Collections/src/System/Collections/BitArray.cs

```diff
@@ -307,10 +309,12 @@ public unsafe BitArray Or(BitArray value)
         {
             if (value == null)
                 throw new ArgumentNullException(nameof(value));
-            if (Length != value.Length)
+
+            int count = Length; // length in bits
```

Comment:
if I understand right, `count` is being used in the same function to represent length in bits, and length in bytes. Instead of the comments, why not use the variable name for clarity? eg
```c#
            int lengthInBits = Length; 
            if (lengthInBits != value.Length)
                throw new ArgumentException(SR.Arg_ArrayLengthsDiffer);

            int lengthInBytes = GetInt32ArrayLengthFromBitLength(count); 
```
Are you reusing `count` because the JIT is not smart enough to reuse a register?

Reply:
Also are you avoiding using `Length` directly as a micro optimization? If so we should probably measure whether it matters.

Reply:
>Are you reusing count because the JIT is not smart enough to reuse a register?

I assumed it wasn't.... but apparently it is smarter than me 😅. Can fix that.

>Also are you avoiding using Length directly as a micro optimization?

Yes... though I doubt it'll matter. Happy to change for the sake of clarity. (Which also will get rid of the `bitCount` variable)
[Disasm of version not using Length](https://sharplab.io/#v2:EYLgxg9gTgpgtADwGwBYA0AXEBDAzgWwB8BiAVwDsATGAMwAIARAUQCEBVAcQFgAoAAQAMdPgEYA3L0HCRAOgBKFDAEt8MGQElyGKEvK4lYXBP5DRMhkuwBzchFzLDxqWYVaVazdt37DMgBoAHEjGkgDMwgBMdADCdLwA3rx0yXQADjoAbtgYMHS6GADaALp0+AD62FBQ2ACexilpmdm5+aVlADYw5FYYABb1KelKWTl5WnRlGTBQ+hDkA8lJg02jkHoYYxtlAMq9OuQA1gAqezC4vRDtlHQAvHQRAKzBS4s8DUMjuWv2m3QsShhcAAFaaeUJRO7ghaNYbNOjfDatFg1HLA0FacG3OgoaEfOEI37/QEgqDI0Z3AIhN7LWGrOY/JEA3ZKGgYElgiF0B64lZfemI8ZE5msklk3KQnm0vnrQlM3osjAAMWgYrRUA5WIi9RecWpyT44VaABkuj1ejrEnqGnQrDAMDqGpbrc7hAB2Nqdbp9aENAC+DpSuDtAeSTpdDRZdAAFFl2qRcgAeOgCACUIetYfDzr6UAgAHc6OQYAWAIJQKykVRaADypAw1Zocmw3RgTAQYBgqWUcyj5GwqggNBj2DjMBTaDosfjE4ARNs5DIyxWq/W6w2my2ygA5GAwShbuY7qxbyszlM+l3+njpiPjIt5/K4LEcO0csvVGomr29RW5/BEr8zWHUdzxvFJI17YtHzoAA+NpKg/GRAL6OhCEIQsoK0J8AGoJl2fZjlOc5LmuJNygQ2okNNPo0ytcNMyzBoAHomJtXMC2gOhzgI6NIzzPBlG6UpoFyPpmzwvZdEI2BiKuTZcFoxjnXfSi5DOJQAC8YCjWB6HIqpagne9H3POgWLoI5qwYas4C3NgjSNEsWCNJgQDoXoMAwVJcBAFirABXpSGAGRIHwJjKAgDAiwwJjc1wdoanIJilFwXB41wJiIiQV0kBEMDkivfK8noYD41gj1qN6RSlIYpSzNYsBOkqdylCsXo6GAAFJxHdKxjoPpcnaATNiK28NiGn47ijcpPTNOg4DoEQU1guChXlEV0QwKFRpSCwMnBNT8GmjpKonCA61+TrAVAuilIgq6nzg1MdtDF7rX0j8CgmjASgAMimkQ6ATJMHuWhbxDewrbsY8zGpgZrYHwbBdF0KxfinM43o+yiS1wbZUmbKNvroXCRCMzDAXmuhifBlMZGiJqoCjG66qhuqZsqrEMYvZ1JmmWZyGw7CeboK8/SSHU8VGUQkF+F8MDfAzP0q38IH/AFkN6KNWnIarXuhxgYGAUgrEXVLpgwXtYLuVMRb4d1ta0FMoyjUh8md8gqcB3Co0B4G/jlBV2QxCIU2WmDVsDjb1RDlmUjFlJJd5aRZdaPaDpgI6dcrYBpjOi7WkR5GqGmPW6FqlI3fGABHUhIqULoNimquMBTcgc+mMy6G2g2i90agoCxdv8FzwffujTFabEZjWJgWvYU9DYMAgQsO8HgBSbuIgnfRyA7Le8ifbA0nzTvB3udN7d4p3a/rxvoQT14GgNOgKFwbAaFyWISyoKNYgxsuFdkgQQxrcO47d2jtDLuGHM+YMKlnLJWRuJ5IFtg7F2JQPY+wDiHAAm66ZWhXWiOdcYdxNbT3MrNFCugOoAlwAQkqRCSEbAAIR3AxlRb80CXSwILPeOgS4kFaDQZ2bs5AoxzgXEuMoKllbflwBYGgn8oBnipOGVokBFDPlfCHWRmtVbqwwJrKMTDFCmXMrYAsVD2o0OACiTG14Da4AfBgMA7UoyaKdumIB1owB4FyKENy2MagFAiH9dhPU1DBNCUUaeVhIorz8UGe4IsGhJNyBEIJFQlYFBEOE7qo4ZDRLyXEhJ8J/GLVSSkdJi0skURCQIfJHDomNNKcvRgcwYBVOSDUgQbl4ntIYJ0kWj9rStCUFiAQIsILbCDBEDQeNSCpFSNAHIlBuH6zqjQJQCA9zXwwAAKmprQNk2gsTBI2dabZuzriO0OXQHQbVTmDwiYUi5RUfHhhoJxKM08JlJk8RsBaUYABqMAwDLygCICIAQEz5BgvTZhXtTITOwncMFELoDQthfCxFZi3qfKUhiyF2K4VaDgp0VkGKsSzJgPMo0EBsCUGJVimFRMTlAjObhJQcc6rOhZVCmFZKMBwUeR5aldxaX0sZcy8FJK2ViueSTPIvK+UNClTIbYkLtKUqVdyicGqf6UHZVS8FE5FUYrDt060ozwy2vjumb5g9fl5CBvCZhfyhaXIaNEpQTTIlFOyZ9P10IhlFhAOmPmMxMGC2Fpfd0fQUoP14L6IAA=)
[Disasm of version using Length](https://sharplab.io/#v2:EYLgxg9gTgpgtADwGwBYA0AXEBDAzgWwB8BiAVwDsATGAMwAIARAUQCEBVAcQFgAoAAQAMdPgEYA3L0HCRAOgBKFDAEt8MGQElyGKEvK4lYXBP5DRMhkuwBzchFzLDxqWYVaVazdt37DMgBoAHEjGkgDMwgBMdADCdLwA3rx0yXQADjoAbtgYMHS6GADaALp0+AD62FBQ2ACexilpmdm5+aVlADYw5FYYABb1KelKWTl5WnRlGTBQ+hDkA8lJg02jkHoYYxtlAMq9OuQA1gAqezC4vRDtlHQAvHQRAKzBS4s8DUMjuWv2m3QsShhcAAFaaeUJRO7ghaNYbNOjfDatFg1HLA0FacG3OgoaEfOEI37/QEgqDI0Z3AIhN7LWGrOY/JEA3ZKGgYElgiF0B64lZfemI8ZE5msklk3KQnm0vnrQlM3osjAAMWgYrRUA5WIi9RecWpyT44VaABkuj1ejrEnqGnQrDAMDqGpbrc7hAB2Nqdbp9aENAC+DpSuDtAeSTpdDRZdAAFFl2qRcgAeOgCACUIetYfDzr6UAgAHc6OQYAWAIJQKykVRaADypAw1Zocmw3RgTAQYBgqWUcyj5GwqggNBj2DjMBTaDosfjE4ARNs5DIyxWq/W6w2my2ygA5GAwShbuY7qxbyszlM+l3+njpiPjIt5/K4LEcO0csvVGomr29RW5/BEr8zWHUdzxvFJI17YtHzoAA+NpKg/GRAL6OhCEIQsoK0J8AGoJl2fZjlOc5LmuJNygQ2okNNPo0ytcNMyzBoAHomJtXMC2gOhzgI6NIzzPBlG6UpoFyPpmzwvZdEI2BiKuTZcFoxjnXfSi5DOJQAC8YCjWB6HIqpagne9H3POgWLoI5qwYas4C3NgjSNEsWCNJgQDoXoMAwVJcBAFirABXpSGAGRIHwJjKAgDAiwwJjc1wdoanIJilFwXB41wJiIiQV0kBEMDkivfK8noYD41gj1qN6RSlIYpSzNYsBOkqdylCsXo6GAAFJxHdKxjoPpcnaATNiK28NiGn47ijcpPTNOg4DoEQU1guChXlEV0QwKFRpSCwMnBNT8GmjpKonCA61+TrAVAuilIgq6nzg1MdtDF7rX0j8CgmjASgAMimkQ6ATJMHuWhbxDewrbsY8zGpgZrYHwbBdF0KxfinM43o+yiS1wbZUmbKNvroXCRCMzDAXmuhifBlMZGiJqoCjG66qhuqZsqrEMYvZ1JmmWZyGw7CeboK8/SSHU8VGUQkF+F8MDfAzP0q38IH/AFkN6KNWnIarXuhxgYGAUgrEXVLpgwXtYLuVMRb4d1ta0FMoyjUh8md8gqcB3Co0B4G/jlBV2QxCIU2WmDVsDjb1RDlmUjFlJJd5aRZdaPaDpgI6dcrYBpjOi7WkR5GqGmPW6FqlI3fGABHUhIqULoNimquMBTcgc+mMy6G2g2i90agoCxdv8FzwffujTFabEZjWJgWvYU9DYMAgQsO8HgBSbuIgnfRyA7Le8ifbA0nzTvB3udN7d4p3a/rxvoQT14GgNOgKFwbAaFyWISyoKNYgxsuFdkgQQxrcO47d2jtDLuGHM+YMKlnLJWRuJ5IFtg7F2JQPY+wDiHAAuOzoIKazoAAQjuBjKi35oEulgQWe8dAlxIK0Ggzs3ZyBRjnAuJcZQVLK2/LgCwNBP5QDPFScMrRICKGfK+EOPDNaq3VhgTWUZNamXMrYAss0UK6A6iiTG14Da4AfBgMA7UowSKdumIB1owB4FyKENy2MagFAiH9MhPU1COOcUUaeVhIorxsUGe4IsGgBNyBEBxFQlYFBEK47qo4ZCeJiT4vx8JbGLWCSkUJi0IkUScQIWJ5DPH5OScvRgcwYAZOSFkgQblfGlIYOUkWj9rStCUFiAQIsILbCDBEDQeNSCpFSNAHIlAqH6zqjQJQCA9zXwwAAKmprQNk2gsSOLGdaSZ0zriO3mXQHQbVlmDzcfEtZRUrHhhoJxKM082lJnMRsBaUYABqMAwDLygCICIAQEz5BgvTc64wp55BJncF5bzoCfO+b8/5ih1n0Teg0MF7zIU/K0HBTorIwVYm6TAXpRoIDYEoEiiFXyiZLKBCs3CSh8F1WtMSj5XzUUYDgvsjyWK7g4rxQSolrzkWktZYckmeQaW0pSJymQ2x3naQxYKqlE5xU/0oGSzFryJwCrBWHSp1pmnhh1fHdMlzB7XOBXcgFGAblCzhSkTxSgCnuISZEz6troQNKLCAdMfMZiYMFsLS+7o+gpQfrwX0QA=)

Reply:
Yeah, we generally only do either of those things when we care enough to measure carefully and prove they really matter, as otherwise we prefer clearer code and making JIT improvements over time.

Reply:
Fair enough. Thanks!